source/store/actions/AuthActions.ts

```diff
@@ -0,0 +1,253 @@
+  return {
+    type: ActionTypes.setAuthStatus,
+    payload: status,
+  };
+}
+
+export function removeProfile(): RemoveProfileDispatch {
+  return {
+    type: ActionTypes.removeUserProfile,
+  };
+}
+
+export function setError(error: DispatchError): SetErrorDispatch {
+  return {
+    type: ActionTypes.setAuthError,
+    payload: error,
+  };
+}
+
+export async function loginFailure(): Promise<LoginFailureDispatch> {
+  await authService.removeAccessTokenFromStorage();
+  return {
+    type: ActionTypes.loginUserFailure,
+  };
+}
+
+export async function addProfile(): Promise<AddProfileDispatch> {
+  const decodedToken = await authService.getAccessTokenFromStorage();
+  const [userProfile, userError] = (await authService.getUserProfile(
+    decodedToken.accessToken
+  )) as [User | null, Error | null];
+
+  if (userError) {
+    throw userError;
+  }
+
+  return {
+    type: ActionTypes.addUserProfile,
+    payload: userProfile,
+  };
+}
+
+export async function refreshSession(): Promise<
+  RefreshSessionSuccessDispatch | RefreshSessionFailureDispatch
+> {
+  const [, refreshError] = await authService.refreshTokens();
+  if (refreshError) {
+    return {
+      type: ActionTypes.authError,
+      payload: refreshError,
+    };
+  }
+  return {
+    type: ActionTypes.refreshUserSession,
+  };
+}
+
+export async function startAuth(
+  personalNumber: string,
+  authenticateOnExternalDevice: boolean
+): Promise<StartAuthSuccessDispatch | StartAuthFailureDispatch> {
+  try {
+    const response = await bankid.auth(personalNumber);
+    if (response.success === false) {
+      throw new Error(response.data);
+    }
+    const { orderRef, autoStartToken } = response.data;
+
+    if (!authenticateOnExternalDevice) {
+      // Start the bankId app on the device for user authorization.
+      await bankid.launchApp(autoStartToken);
+    }
+
+    return {
+      type: ActionTypes.authStarted,
+      payload: {
+        orderRef,
+        autoStartToken,
+      },
+    };
+  } catch (error) {
+    return {
+      type: ActionTypes.authError,
+      payload: error as DispatchError,
+    };
+  }
+}
+
+export async function startSign(
+  personalNumber: string,
+  userVisibleData: string,
+  authenticateOnExternalDevice: boolean
+): Promise<StartSignSuccessDispatch | StartSignFailureDispatch> {
+  try {
+    const response = await bankid.sign(personalNumber, userVisibleData);
+
+    if (response.success === false) {
+      throw new Error(response.data);
+    }
+
+    const { orderRef, autoStartToken } = response.data;
+
+    if (!authenticateOnExternalDevice) {
+      // Start the bankId app on the device for user authorization.
+      await bankid.launchApp(autoStartToken);
+    }
+
+    return {
+      type: ActionTypes.signStarted,
+      payload: {
+        orderRef,
+        autoStartToken,
+      },
+    };
+  } catch (error) {
+    return {
+      type: ActionTypes.signFailure,
+      payload: error?.data ?? null,
```

Comment:
Fix this somehow. `error?.data` _can_ be a `string` (e.g. check `bankid.sign(...)`).

Reply:
added string type